Assets/Stack mini game/Scripts/GameState.cs

```diff
@@ -8,17 +8,26 @@
 [Serializable]
 public class GameState
 {
-    [SerializeField] private StackMiniGame_StateRegister stateRegister;
+    [SerializeField] private StateRegister stateRegister;
 
-    public string CurrentState { get; private set; }
+    private string currentState;
 
     public void SetCurrentState(string newState)
     {
         if(TryGetState(newState.ToLower().Trim(), out string result) == true)
-                CurrentState = result;
+                currentState = result;
     }
 
-    public bool TryGetState(string state, out string result)
+    public bool CheckCurrentState(string state)

```

Comment:
```csharp
    public bool CheckCurrentState(string state)
    {
        return TryGetState(state, out string result) && currentState == result;
    }
```

You should be able to just do this rather than having that redundant string variable

Reply:
Right.